core-plugins/src/test/java/co/cask/hydrator/plugin/batch/source/XMLReaderBatchSourceTest.java

```diff
@@ -0,0 +1,425 @@
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.api.dataset.lib.KeyValueTable;
+import co.cask.cdap.api.dataset.table.Table;
+import co.cask.cdap.etl.api.batch.BatchSource;
+import co.cask.cdap.etl.batch.ETLBatchApplication;
+import co.cask.cdap.etl.batch.mapreduce.ETLMapReduce;
+import co.cask.cdap.etl.mock.batch.MockSink;
+import co.cask.cdap.etl.mock.test.HydratorTestBase;
+import co.cask.cdap.etl.proto.v2.ETLBatchConfig;
+import co.cask.cdap.etl.proto.v2.ETLPlugin;
+import co.cask.cdap.etl.proto.v2.ETLStage;
+import co.cask.cdap.proto.artifact.AppRequest;
+import co.cask.cdap.proto.artifact.ArtifactSummary;
+import co.cask.cdap.proto.id.ApplicationId;
+import co.cask.cdap.proto.id.ArtifactId;
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.test.ApplicationManager;
+import co.cask.cdap.test.DataSetManager;
+import co.cask.cdap.test.MapReduceManager;
+import co.cask.cdap.test.TestConfiguration;
+import co.cask.hydrator.common.Constants;
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.python.google.common.collect.ImmutableMap;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Unit test for {@link XMLReaderBatchSource} class.
+ */
+public class XMLReaderBatchSourceTest extends HydratorTestBase {
+
+  private static final ArtifactVersion CURRENT_VERSION = new ArtifactVersion("3.4.0-SNAPSHOT");
+  private static final ArtifactId BATCH_APP_ARTIFACT_ID =
+    NamespaceId.DEFAULT.artifact("etlbatch", CURRENT_VERSION.getVersion());
+  private static final ArtifactSummary ETLBATCH_ARTIFACT =
+    new ArtifactSummary(BATCH_APP_ARTIFACT_ID.getArtifact(), BATCH_APP_ARTIFACT_ID.getVersion());
+  private static final String SOURCE_FOLDER_PATH = "src/test/resources/xmlsource/";
+  private File sourceFolder = new File(SOURCE_FOLDER_PATH);
+  private File targetFolder = new File("src/test/resources/xmltarget/");
+  private String sourceFolderUri = sourceFolder.toURI().toString();
+  private String targetFolderUri = targetFolder.toURI().toString();
+
+  @ClassRule
+  public static final TestConfiguration CONFIG = new TestConfiguration("explore.enabled", false);
+
+  @BeforeClass
+  public static void setupTest() throws Exception {
+    setupBatchArtifacts(BATCH_APP_ARTIFACT_ID, ETLBatchApplication.class);
+    addPluginArtifact(NamespaceId.DEFAULT.artifact("core-plugins", "1.0.1"), BATCH_APP_ARTIFACT_ID,
+                      XMLReaderBatchSource.class);
+  }
+
+  /**
+   * Method to copy test xml files into source folder path, from where XMLReader read the file.
+   */
+  @Before
+  public void copyFiles() throws IOException {
+      String xmlSourceFolder = "src/test/resources/";
```

Comment:
it is working on Windows. 
I have to copy this files after every test case run because test case may delete, move or archive these files, so need to be copied again for next test case run.